kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h

```diff
@@ -23,97 +23,105 @@
 
 namespace Kratos
 {
-    template<int TDimension, class TPointType> class NurbsCurveGeometry;
-    template<int TDimension, class TPointType> class NurbsSurfaceGeometry;
-    namespace ProjectionNurbsGeometryUtilities
+template<class TPointType> class Geometry;
+template<int TDimension, class TPointType> class NurbsSurfaceGeometry;
+namespace ProjectionNurbsGeometryUtilities
+{
```

Comment:
We discussed the addition of extra namespaces in the @KratosMultiphysics/technical-committee and we agreed that this has to be avoided and should be only considered if there is no technical alternative. In this case, the exact same behavior can be obtained using a class with two `static` methods.

About the forward declaration, it should be always avoided. Did you get a recursively include pattern in here?

Reply:
This is actually an older file, just the indentation is changed. However, probably good that we came across it again, then.

Well changing it to a class with static functions would be totally fine (to be honest I don't really know the advantage or disadvantage of it.).

About the forward declaration, as far as I remember this is coming from the issue that this function is being called from within a geometry object. Accordingly, the forward declaration was needed. Do you have any idea to avoid this?
Would a split into .h and .cpp avoid this?

Reply:
Thanks Ruben for the issue. I have created  #8598 to address and discuss this issue.